rotors_driver/src/rotors_driver_node.py

```diff
@@ -0,0 +1,356 @@
+                           capacity=float('nan'), design_capacity=float('nan'), percentage=bat_percent,
+                           power_supply_status=0, power_supply_health=0, power_supply_technology=0, present=True,
+                           cell_voltage=[float('nan')], location="0",
+                           serial_number="")
+		self.battery_state_publisher.publish(bat)
+
+
+		pose = PoseStamped(pose=Pose(position=Point(x=self.current_x, y=self.current_y, z=self.current_z),
+                                     orientation=Quaternion(x=float(self.quaternion[0]), y=float(self.quaternion[1]), z=float(self.quaternion[2]),
+                                                            w=float(self.quaternion[3]))),header = Header(frame_id = self.frame_id))
+		self.rqt_pose_publisher.publish(pose)
+
+
+		# Empty, global pos not known
+		nav_sat = NavSatFix(status=NavSatStatus(status=-1, service=0), latitude=float('nan'),
+                            longitude=float('nan'), altitude=float('nan'), position_covariance=[float('nan')] * 9,
+                            position_covariance_type=0)
+		self.rqt_global_position_publisher.publish(nav_sat)
+
+
+		twist = TwistStamped(twist=Twist(linear=Vector3(x=self.current_vx, y=self.current_vy, z=self.current_vz),
+                                         angular=Vector3(x=self.current_ang_vx, y=self.current_ang_vy, z=self.current_ang_vz)),
+										 header = Header(frame_id = self.frame_id))
+		self.rqt_velocity_body_publisher.publish(twist)
+
+
+		self.rqt_cam_frontal_publisher.publish(self.frontal_image)
+		self.rqt_cam_ventral_publisher.publish(self.ventral_image)
+
+
+	def cam_frontal(self, msg):
+		self.frontal_image = msg
+		rospy.logdebug('Frontal image updated')
+
+	def cam_ventral(self, msg):
+		self.ventral_image = msg
+		rospy.logdebug('Ventral image updated')
+
+	def get_pose(self,msg):
+		self.current_state = msg
+		
+		self.current_x=self.current_state.pose.pose.position.x
+		self.current_y=self.current_state.pose.pose.position.y
+		self.current_z=self.current_state.pose.pose.position.z
+		self.current_vx=self.current_state.twist.twist.linear.x
+		self.current_vy=self.current_state.twist.twist.linear.y
+		self.current_vz=self.current_state.twist.twist.linear.z
+		self.current_ang_vx=self.current_state.twist.twist.angular.x
+		self.current_ang_vy=self.current_state.twist.twist.angular.y
+		self.current_ang_vz=self.current_state.twist.twist.angular.z
+		self.quaternion = [self.current_state.pose.pose.orientation.x, self.current_state.pose.pose.orientation.y,
+					self.current_state.pose.pose.orientation.z, self.current_state.pose.pose.orientation.w
+				]
+		self.current_roll, self.current_pitch, self.current_yaw = tf.transformations.euler_from_quaternion(self.quaternion)
+		self.current_yaw_rate = self.current_state.twist.twist.angular.z
+
+		self.frame_id = self.current_state.header.frame_id
+
+		if self.current_z<0.1:
```

Comment:
Not bad, but it can be done better as finite state machine. Called from arming and takeoff/land methods

Reply:
solved in latest commits